dataset/datasets.py

```diff
@@ -392,9 +484,15 @@ def hermit_polynomial(treatment, gams):
         return y, noise
 
     elif dataset == "tcga-B":  # TCGA modification in SCIGAN (Bica et al., 2020)
-        y = D["y"]
+        V = D["V"]
+        x = x.numpy()
+        t = t.numpy()
 
-        return y
+        y = 10 * (np.dot(V[0], x.T) + (np.dot((12.0 * V[1]), x.T) *  t) - (np.dot((12.0 * V[1]), x.T) *  (t ** 2)))
+
+        noise = np.random.normal(0, 0.2, size = len(y))
```

Comment:
Your previous comment said that noise was none, but here I see that you did return the noise. I will replace this line with         if noise is None:
            noise = noise_scale * torch.randn_like(t)
 for consistency with other datasets.
 Higher noise/error reduces confounding. So controlling it is useful.